src/nbl/asset/interchange/CGLTFLoader.cpp

```diff
@@ -1453,8 +1454,92 @@ namespace nbl
 							pushConstants.availableTextures |= CGLTFPipelineMetadata::SGLTFMaterialParameters::EGT_NORMAL_TEXTURE;
 
 							auto cpuNormalTexture = IMAGE_VIEWS[SGLTF::SGLTFMaterial::EGT_NORMAL_TEXTURE];
+							{
+								core::smart_refctd_ptr<ICPUImage> newDerivativeNormalMapImage;
+
+								auto cpuImageNormalTexture = cpuNormalTexture->getCreationParameters().image;
+
+								const auto referenceImageParams = cpuImageNormalTexture->getCreationParameters();
+								const auto referenceBuffer = cpuImageNormalTexture->getBuffer();
+								const auto referenceRegions = cpuImageNormalTexture->getRegions();
+								const auto* referenceRegion = referenceRegions.begin();
+
+								auto newImageParams = referenceImageParams;
+								core::smart_refctd_ptr<ICPUBuffer> newCpuBuffer;
+
+								newImageParams.format = EF_R8G8_SNORM;
+
+								auto newRegions = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<ICPUImage::SBufferCopy>>(referenceRegions.size());
+
+								size_t regionOffsets = {};
+
+								for (auto newRegion = newRegions->begin(); newRegion != newRegions->end(); ++newRegion)
+								{
+									auto idOffset = newRegion - newRegions->begin();
+									*newRegion = *(referenceRegion++);
+									newRegion->bufferOffset = regionOffsets;
+
+									const auto fullMipMapExtent = cpuImageNormalTexture->getMipSize(idOffset);
```

Comment:
region number has nothing to do with mipmap number

Reply:
whole this code assumes that 1 region = 1 mip level and that they're sorted in increasing order, it's totally wrong

Reply:
I know, hmm.. may I just execute flatten on input image first and then do above assumption?

Reply:
and do not take mip id from `idOffset`

Reply:
corrected in following commit